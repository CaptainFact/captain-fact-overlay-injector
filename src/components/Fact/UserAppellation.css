.container > * {
    text-decoration: underline;
    color: #0095ff;
}