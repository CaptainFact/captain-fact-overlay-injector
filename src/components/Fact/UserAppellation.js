import React from 'react'

import styles from './UserAppellation.module.css'
import { FRONTEND_URL } from '../../constants'

export const UserAppellation = ({ user: { username, name } }) => {
  return (
    <a
      href={`${FRONTEND_URL}/u/${username}`}
      className={styles.container}
      title={name}
    >
      <strong>{`@${username}`}</strong>
    </a>
  )
}

export default UserAppellation
