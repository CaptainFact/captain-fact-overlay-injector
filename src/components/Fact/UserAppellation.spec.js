import React from 'react'
import { UserAppellation } from './UserAppellation'

test('render with only username', () => {
  snapshot(<UserAppellation user={{ username: 'Toto' }} />)
})

test('render with username + name', () => {
  snapshot(<UserAppellation user={{ username: 'Toto', name: 'Titi Toto' }} />)
})
