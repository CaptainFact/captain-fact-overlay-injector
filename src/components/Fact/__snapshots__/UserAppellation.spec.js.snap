// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with only username 1`] = `
<a
  className="container"
  href="https://testing.captainfact.io/u/Toto"
>
  <strong>
    @Toto
  </strong>
</a>
`;

exports[`render with username + name 1`] = `
<a
  className="container"
  href="https://testing.captainfact.io/u/Toto"
  title="Titi Toto"
>
  <strong>
    @Toto
  </strong>
</a>
`;
